### ISSUE TITLE
Block from parity: Import Failure InvalidBlockFormat

### ISSUE THREAD

**Issue by 5chdn** [ethereum/webthree-umbrella#466]
This morning i had a big red warning:

```
  ℹ  06:19:43.977|eth  SLOW IMPORT: #340035c0…  # 1375225
  ℹ  06:19:43.977|eth    Import took: 0.831801
  ℹ  06:19:43.977|eth    preliminaryChecks: 0.000708
  ℹ  06:19:43.977|eth    enactment: 0.828189
  ℹ  06:19:43.977|eth    collation: 5e-06
  ℹ  06:19:43.977|eth    writing: 0.000336
  ℹ  06:19:43.977|eth    checkBest: 0.002483
  ℹ  06:19:43.977|eth    110  transactions
  ℹ  06:19:43.977|eth    4663635  gas used
  ✘  07:48:38.238|p2p|c97f8d74…|Parity/v1.1.0-unstable-dca67a3-20160413/x86_64-linux-gnu/rustc1.7.0-beta.3  


    Import Failure     InvalidBlockFormat                                       

                       Guru Meditation #00000000.8e20de73…                      


  ✘  07:48:38.238|p2p|c97f8d74…|Parity/v1.1.0-unstable-dca67a3-20160413/x86_64-linux-gnu/rustc1.7.0-beta.3  Ignoring malformed block: /home/user/.cache/pacaur/ethereum/src/ethereum/libethereum/libethcore/BlockHeader.cpp(109): Throw in function static dev::RLP dev::eth::BlockHeader::extractHeader(dev::bytesConstRef)
Dynamic exception type: boost::exception_detail::clone_impl<dev::eth::InvalidBlockFormat>
std::exception::what: InvalidBlockFormat
[dev::tag_comment*] = Block transactions must be a list
[dev::eth::tag_now*] = 1461217718
[dev::eth::tag_data*] = �
[dev::eth::tag_block*] = [ type: std::vector<unsigned char, std::allocator<unsigned char> >, size: 24, dump: 30 0c 18 3f 0c 7f 00 00 44 0e 18 3f 0c 7f 00 00 ]
[dev::eth::tag_field*] = 1
[dev::eth::tag_phase*] = 1
```

This is `eth` version 1.2.4 running on arch linux. Not sure if this is a parity issue, but since eth stopped working after that, I want to raise it here first.

Edit: Also note the timestamps. I wonder what the client did in between 06:19 and 07:48. Unfortunately, I've only run it on verbosity level 3. Is there anything else I can provide?

**Comment by bobsummerwill** [comment]
Yikes - thanks for the report!

**Comment by chriseth** [comment]
Behaviour seems to be correct from the yellow paper. Also I assume that this does not happen on import but rather only in the p2p communication with parity. If you can get the full hex data of the block, I can take a deeper look, but otherwise I would just assume this is a bug in parity.

**Comment by jesse-git** [comment]
Newcomer to ethereum here, but I received same error on v1.2.4 Win7. I was using AlethZero and only enabled network communication and then Network->Go. Debug level was also only set at 1. Hours later, received:

```
  i  15:38:10.477|eth  SLOW IMPORT: #e4ecee40ΓÇª  # 1469370
  i  15:38:10.479|eth    Import took: 0.513029
  i  15:38:10.480|eth    preliminaryChecks: 0.001
  i  15:38:10.482|eth    enactment: 0.510029
  i  15:38:10.484|eth    collation: 0
  i  15:38:10.485|eth    writing: 0
  i  15:38:10.486|eth    checkBest: 0
  i  15:38:10.487|eth    21  transactions
  i  15:38:10.489|eth    511735  gas used
  X  15:42:29.193|p2p|c97f8d74ΓÇª|Parity/v1.1.0-unstable-dca67a3-20160413/x86_64-linux-gnu/rustc1.7.0-beta.3

    Import Failure     InvalidBlockFormat

                       Guru Meditation #00000000.8e20de73ΓÇª

  X  15:42:29.212|p2p|c97f8d74ΓÇª|Parity/v1.1.0-unstable-dca67a3-20160413/x86_64
-linux-gnu/rustc1.7.0-beta.3  Ignoring malformed block: C:\Jenkins\sharedspace\ethrelease\label\windows-slave\webthree-umbrella\libethereum\libethcore\BlockHead
er.cpp(109): Throw in function class dev::RLP __cdecl dev::eth::BlockHeader::extractHeader(class dev::vector_ref<unsigned char const >)
Dynamic exception type: class boost::exception_detail::clone_impl<struct ev::eth::InvalidBlockFormat>
std::exception::what: InvalidBlockFormat
[struct dev::eth::tag_block * __ptr64] = [ type: class std::vector<unsigned char,class std::allocator<unsigned char> >, size: 24, dump: 00 65 74 1a 00 00 00 00
1d 67 74 1a 00 00 00 00 ]
[struct dev::tag_comment * __ptr64] = Block transactions must be a list
[struct dev::eth::tag_data * __ptr64] = Ç
[struct dev::eth::tag_field * __ptr64] = 1
[struct dev::eth::tag_now * __ptr64] = 1462563749
[struct dev::eth::tag_phase * __ptr64] = 1
```

Happens every time I restart the app. Is it safe to ignore?

**Comment by 5chdn** [comment]
While I was not able to reproduce this again at a later point, I would like to emphasize that my eth node actually _stopped_ working after that exception.

**Comment by chriseth** [comment]
@5chdn thanks for the info, I get those errors occasionally, so I'll keep an eye open.

**Comment by 5chdn** [comment]
I would like to escalate this issue to a higher priority. This issue is now reproducable on my machine and basicly prevents `eth` from syncing the blockchain.

```
 ~ $ eth --listen 30304 --no-ipc
cpp-ethereum, a C++ Ethereum client
cpp-ethereum 1.2.4
  By cpp-ethereum contributors, (c) 2013-2016.
  See the README for contributors and credits.
Transaction Signer: XE970LFJ45WXCWXB6AJBEMXUYMVX33V7P5 (bb63f71d-235a-cef6-f1d1-5a0807a058e6 - 00243db9)
Mining Beneficiary: XE970LFJ45WXCWXB6AJBEMXUYMVX33V7P5 (bb63f71d-235a-cef6-f1d1-5a0807a058e6 - 00243db9)
Foundation: XE55PXQKKK4B9BYPBGT1XCYW6R5ELFAT6EM (00000000-0000-0000-0000-000000000000 - de0b2956)
  ℹ  08:57:49.888|p2p  UPnP device: http://192.168.178.1:49000/igddesc.xml [st: urn:schemas-upnp-org:device:InternetGatewayDevice:1 ]
Node ID: enode://6fb7872da1f2fd3820f8888315f806ffb1ef72449f772c50543a1c3aac4ae530ef467f42b280c9c11127b065fa4b6cd8c756a1bcd81a3dd9d5b59fe5ba24c362@0.0.0.0:0
  ✘  09:18:35.260|p2p|579beb09…|Parity/v1.1.0-unstable-0b78a1e-20160424/x86_64-linux-gnu/rustc1.8.0-beta.1  


    Import Failure     InvalidBlockFormat                                       

                       Guru Meditation #00000000.8e20de73…                      


  ✘  09:18:35.260|p2p|579beb09…|Parity/v1.1.0-unstable-0b78a1e-20160424/x86_64-linux-gnu/rustc1.8.0-beta.1  Ignoring malformed block: /home/user/.cache/pacaur/ethereum/src/ethereum/libethereum/libethcore/BlockHeader.cpp(109): Throw in function static dev::RLP dev::eth::BlockHeader::extractHeader(dev::bytesConstRef)
Dynamic exception type: boost::exception_detail::clone_impl<dev::eth::InvalidBlockFormat>
std::exception::what: InvalidBlockFormat
[dev::tag_comment*] = Block transactions must be a list
[dev::eth::tag_now*] = 1462864715
[dev::eth::tag_data*] = �
[dev::eth::tag_block*] = [ type: std::vector<unsigned char, std::allocator<unsigned char> >, size: 24, dump: 50 69 ad 31 ba 7f 00 00 6d 6b ad 31 ba 7f 00 00 ]
[dev::eth::tag_field*] = 1
[dev::eth::tag_phase*] = 1

  ✘  09:22:42.368|p2p|281c7628…|Parity/v1.1.0-beta/x86_64-linux-gnu/rustc1.8.0  


    Import Failure     InvalidBlockFormat                                       

                       Guru Meditation #00000000.8e20de73…                      


  ✘  09:22:42.369|p2p|281c7628…|Parity/v1.1.0-beta/x86_64-linux-gnu/rustc1.8.0  Ignoring malformed block: /home/user/.cache/pacaur/ethereum/src/ethereum/libethereum/libethcore/BlockHeader.cpp(109): Throw in function static dev::RLP dev::eth::BlockHeader::extractHeader(dev::bytesConstRef)
Dynamic exception type: boost::exception_detail::clone_impl<dev::eth::InvalidBlockFormat>
std::exception::what: InvalidBlockFormat
[dev::tag_comment*] = Block transactions must be a list
[dev::eth::tag_now*] = 1462864962
[dev::eth::tag_data*] = �
[dev::eth::tag_block*] = [ type: std::vector<unsigned char, std::allocator<unsigned char> >, size: 24, dump: b0 c5 1c 30 ba 7f 00 00 cd c7 1c 30 ba 7f 00 00 ]
[dev::eth::tag_field*] = 1
[dev::eth::tag_phase*] = 1

  ✘  09:22:54.104|p2p|08ff5430…|Parity/v1.0.0-beta/x86_64-linux-gnu/rustc1.9.0-nightly  


    Import Failure     InvalidBlockFormat                                       

                       Guru Meditation #00000000.8e20de73…                      


  ✘  09:22:54.104|p2p|08ff5430…|Parity/v1.0.0-beta/x86_64-linux-gnu/rustc1.9.0-nightly  Ignoring malformed block: /home/user/.cache/pacaur/ethereum/src/ethereum/libethereum/libethcore/BlockHeader.cpp(109): Throw in function static dev::RLP dev::eth::BlockHeader::extractHeader(dev::bytesConstRef)
Dynamic exception type: boost::exception_detail::clone_impl<dev::eth::InvalidBlockFormat>
std::exception::what: InvalidBlockFormat
[dev::tag_comment*] = Block transactions must be a list
[dev::eth::tag_now*] = 1462864974
[dev::eth::tag_data*] = �
[dev::eth::tag_block*] = [ type: std::vector<unsigned char, std::allocator<unsigned char> >, size: 24, dump: b0 c5 1c 30 ba 7f 00 00 cd c7 1c 30 ba 7f 00 00 ]
[dev::eth::tag_field*] = 1
[dev::eth::tag_phase*] = 1

  ✘  09:48:35.955|p2p|08ff5430…|Parity/v1.0.0-beta/x86_64-linux-gnu/rustc1.9.0-nightly  


    Import Failure     InvalidBlockFormat                                       

                       Guru Meditation #00000000.8e20de73…                      


  ✘  09:48:35.955|p2p|08ff5430…|Parity/v1.0.0-beta/x86_64-linux-gnu/rustc1.9.0-nightly  Ignoring malformed block: /home/user/.cache/pacaur/ethereum/src/ethereum/libethereum/libethcore/BlockHeader.cpp(109): Throw in function static dev::RLP dev::eth::BlockHeader::extractHeader(dev::bytesConstRef)
Dynamic exception type: boost::exception_detail::clone_impl<dev::eth::InvalidBlockFormat>
std::exception::what: InvalidBlockFormat
[dev::tag_comment*] = Block transactions must be a list
[dev::eth::tag_now*] = 1462866515
[dev::eth::tag_data*] = �
[dev::eth::tag_block*] = [ type: std::vector<unsigned char, std::allocator<unsigned char> >, size: 24, dump: e0 52 24 30 ba 7f 00 00 fd 54 24 30 ba 7f 00 00 ]
[dev::eth::tag_field*] = 1
[dev::eth::tag_phase*] = 1

  ✘  10:19:01.847|p2p|5d7684c6…|Parity/v1.1.0-beta/x86_64-linux-gnu/rustc1.8.0  


    Import Failure     InvalidBlockFormat                                       

                       Guru Meditation #00000000.8e20de73…                      


  ✘  10:19:01.847|p2p|5d7684c6…|Parity/v1.1.0-beta/x86_64-linux-gnu/rustc1.8.0  Ignoring malformed block: /home/user/.cache/pacaur/ethereum/src/ethereum/libethereum/libethcore/BlockHeader.cpp(109): Throw in function static dev::RLP dev::eth::BlockHeader::extractHeader(dev::bytesConstRef)
Dynamic exception type: boost::exception_detail::clone_impl<dev::eth::InvalidBlockFormat>
std::exception::what: InvalidBlockFormat
[dev::tag_comment*] = Block transactions must be a list
[dev::eth::tag_now*] = 1462868341
[dev::eth::tag_data*] = �
[dev::eth::tag_block*] = [ type: std::vector<unsigned char, std::allocator<unsigned char> >, size: 24, dump: c0 3f c6 30 ba 7f 00 00 dd 41 c6 30 ba 7f 00 00 ]
[dev::eth::tag_field*] = 1
[dev::eth::tag_phase*] = 1

  ℹ  10:29:55.687|eth  SLOW IMPORT: #4cf49cba…  # 1490722
  ℹ  10:29:55.687|eth    Import took: 0.613248
  ℹ  10:29:55.687|eth    preliminaryChecks: 0.001136
  ℹ  10:29:55.687|eth    enactment: 0.585642
  ℹ  10:29:55.687|eth    collation: 0.02054
  ℹ  10:29:55.687|eth    writing: 0.00018
  ℹ  10:29:55.687|eth    checkBest: 0.005687
  ℹ  10:29:55.687|eth    219  transactions
  ℹ  10:29:55.687|eth    4644331  gas used
  ℹ  10:30:42.598|eth  SLOW IMPORT: #25aa5dc1…  # 1490723
  ℹ  10:30:42.598|eth    Import took: 0.622587
  ℹ  10:30:42.598|eth    preliminaryChecks: 0.001116
  ℹ  10:30:42.598|eth    enactment: 0.618087
  ℹ  10:30:42.598|eth    collation: 3e-06
  ℹ  10:30:42.598|eth    writing: 0.000141
  ℹ  10:30:42.598|eth    checkBest: 0.003194
  ℹ  10:30:42.598|eth    220  transactions
  ℹ  10:30:42.598|eth    4663038  gas used
  ℹ  10:31:44.515|eth  SLOW IMPORT: #3aa43569…  # 1490725
  ℹ  10:31:44.515|eth    Import took: 0.554965
  ℹ  10:31:44.515|eth    preliminaryChecks: 0.001079
  ℹ  10:31:44.516|eth    enactment: 0.551062
  ℹ  10:31:44.516|eth    collation: 3e-06
  ℹ  10:31:44.516|eth    writing: 0.000143
  ℹ  10:31:44.516|eth    checkBest: 0.00263
  ℹ  10:31:44.516|eth    210  transactions
  ℹ  10:31:44.516|eth    4654088  gas used
  ℹ  10:32:06.593|eth  SLOW IMPORT: #f1714773…  # 1490726
  ℹ  10:32:06.593|eth    Import took: 0.586675
  ℹ  10:32:06.593|eth    preliminaryChecks: 0.001114
  ℹ  10:32:06.593|eth    enactment: 0.58249
  ℹ  10:32:06.593|eth    collation: 3e-06
  ℹ  10:32:06.593|eth    writing: 0.000151
  ℹ  10:32:06.593|eth    checkBest: 0.002866
  ℹ  10:32:06.593|eth    220  transactions
  ℹ  10:32:06.593|eth    4706336  gas used
  ℹ  10:34:13.862|eth  SLOW IMPORT: #13daf95a…  # 1490730
  ℹ  10:34:13.862|eth    Import took: 0.673277
  ℹ  10:34:13.863|eth    preliminaryChecks: 0.001188
  ℹ  10:34:13.863|eth    enactment: 0.668861
  ℹ  10:34:13.863|eth    collation: 3e-06
  ℹ  10:34:13.863|eth    writing: 0.000139
  ℹ  10:34:13.863|eth    checkBest: 0.003037
  ℹ  10:34:13.863|eth    224  transactions
  ℹ  10:34:13.863|eth    4712244  gas used
  ✘  10:48:41.032|p2p|281c7628…|Parity/v1.1.0-beta/x86_64-linux-gnu/rustc1.8.0  


    Import Failure     InvalidBlockFormat                                       

                       Guru Meditation #00000000.8e20de73…                      


  ✘  10:48:41.032|p2p|281c7628…|Parity/v1.1.0-beta/x86_64-linux-gnu/rustc1.8.0  Ignoring malformed block: /home/user/.cache/pacaur/ethereum/src/ethereum/libethereum/libethcore/BlockHeader.cpp(109): Throw in function static dev::RLP dev::eth::BlockHeader::extractHeader(dev::bytesConstRef)
Dynamic exception type: boost::exception_detail::clone_impl<dev::eth::InvalidBlockFormat>
std::exception::what: InvalidBlockFormat
[dev::tag_comment*] = Block transactions must be a list
[dev::eth::tag_now*] = 1462870121
[dev::eth::tag_data*] = �
[dev::eth::tag_block*] = [ type: std::vector<unsigned char, std::allocator<unsigned char> >, size: 24, dump: 50 a6 ad 30 ba 7f 00 00 6d a8 ad 30 ba 7f 00 00 ]
[dev::eth::tag_field*] = 1
[dev::eth::tag_phase*] = 1

  ✘  11:14:05.717|p2p|579beb09…|Parity/v1.1.0-unstable-0b78a1e-20160424/x86_64-linux-gnu/rustc1.8.0-beta.1  


    Import Failure     InvalidBlockFormat                                       

                       Guru Meditation #00000000.8e20de73…                      


  ✘  11:14:05.717|p2p|579beb09…|Parity/v1.1.0-unstable-0b78a1e-20160424/x86_64-linux-gnu/rustc1.8.0-beta.1  Ignoring malformed block: /home/user/.cache/pacaur/ethereum/src/ethereum/libethereum/libethcore/BlockHeader.cpp(109): Throw in function static dev::RLP dev::eth::BlockHeader::extractHeader(dev::bytesConstRef)
Dynamic exception type: boost::exception_detail::clone_impl<dev::eth::InvalidBlockFormat>
std::exception::what: InvalidBlockFormat
[dev::tag_comment*] = Block transactions must be a list
[dev::eth::tag_now*] = 1462871645
[dev::eth::tag_data*] = �
[dev::eth::tag_block*] = [ type: std::vector<unsigned char, std::allocator<unsigned char> >, size: 24, dump: 40 3a 6c 30 ba 7f 00 00 5f 3c 6c 30 ba 7f 00 00 ]
[dev::eth::tag_field*] = 1
[dev::eth::tag_phase*] = 1
```

If you could guide me how to extract the full hex data of the malformed block, I could try to get more details.

**Comment by winsvega** [comment]
@5chdn you mean it prevents eth from syncing by continuously repeating error messages?

just reproduced bad rlp block import issue on ubuntu eth. it wont make eth stop syncing or anything. 
and as I could see in your logs, your eth continue syncing as well. 
it seems to be the Parity issue.

**Comment by 5chdn** [comment]
After running in this mode for some more days it seems like it somehow does pick up the sync again at a certain point. I am wondering if its normal that eth is not printing anything to standard output for several hours?

**Comment by winsvega** [comment]
@5chdn it is fine, you could always check the progress of importing blocks with mist running parallel.

but as I noticed such case happen at the beginning when blocks processed fast.
from the middle of syncing there are always SLOW IMPORT logs appear at the console

**Comment by 5chdn** [comment]
Okay, I think this can be closed as _works as expected_. Maybe I point some parity developers to this issue. Thanks for your feedback.

**Comment by jderocher** [comment]
New miner here. I just installed ethminer today. FWIW, I first tried using geth, geth worked fine, I then learned geth didn't support GPU mining (I found this odd as a newcomer), I quickly installed cpp-ethereum and tried using ethminer. I get an unlimited amount of blocks that read as (see 1* below).

Also, if I try doing eth -M -G (even with the 20 sec benchmark delay, it recognizes my suitable openCL GPU, but then just spits this out every time (see 2** below)

Errors produced on Windows 10, z87 Asus 1150 nvidia gtx 1070. eth version 1.2.9 npv: 63, cdv: 12041

What is going on? some people have said they've just waited and it has gone away. I've never had a successful mining session and I honestly am a bit overwhelmed but this github issue is the only place I even saw my issue posted anywhere on the net and I don't know where to begin to ask for support. Thanks for your time guys.

Error 1*:

X  20:19:41.009|txcheck1  Bad transaction: C:\Jenkins\sharedspace\ethrelease\label\windows-slave\webthree-umbrella\libethereum\libethcore\Transaction.cpp(76): Throw in function __cdecl dev::eth::TransactionBase::TransactionBase(class dev::vector_ref<unsigned char const >,enum dev::eth::CheckTransaction)
Dynamic exception type: class boost::exception_detail::clone_impl<struct dev::eth::InvalidSignature>
std::exception::what: InvalidSignature
[struct dev::eth::tag_data * __ptr64] = a07410a6f8ab5870f6f334b2c10ea515acb79a984ff22e0b17cb3fd4b602fff019
[struct dev::eth::tag_field * __ptr64] = 8
[struct dev::eth::tag_field * __ptr64] = invalid transaction format


Error 2**:

C:\Program Files (x86)\cpp-ethereum>eth -M -G
cpp-ethereum, a C++ Ethereum client
Found suitable OpenCL device [GeForce GTX 1070] with 8589934592 bytes of GPU memory
Benchmarking on platform: { "platform": "NVIDIA CUDA", "device": "GeForce GTX 1070", "version": "OpenCL 1.2 CUDA" }
Preparing DAG...
  i  20W:a2r7m:i1n3g. 1u5p5.|.g.p
uminer0  workLoop 0 #00000000ΓÇª #00000000ΓÇª
  i  20:27:13.157|gpuminer0  Initialising miner...
Trial 1... 0
Trial 2... 0
Trial 3... 0
Trial 4... 0
Trial 5... 0
